tests/unit/test_flask_app.py

```diff
@@ -43,3 +44,48 @@ def test_flask_env(harness: Harness, flask_config: dict):
         f"{FLASK_ENV_CONFIG_PREFIX}{k.upper()}": v if isinstance(v, str) else json.dumps(v)
         for k, v in flask_config.items()
     }
+
+
+@pytest.mark.parametrize(
+    "set_environment_variable, expected",
+    [
+        pytest.param(
+            {"JUJU_CHARM_NO_PROXY": "127.0.0.1"}, {"no_proxy": "127.0.0.1"}, id="no_proxy"
+        ),
+        pytest.param(
+            {"JUJU_CHARM_HTTP_PROXY": "proxy.test"}, {"http_proxy": "proxy.test"}, id="http_proxy"
+        ),
+        pytest.param(
+            {"JUJU_CHARM_HTTPS_PROXY": "proxy.test"},
+            {"https_proxy": "proxy.test"},
+            id="https_proxy",
+        ),
+        pytest.param(
+            {"JUJU_CHARM_HTTP_PROXY": "proxy.test", "JUJU_CHARM_HTTPS_PROXY": "proxy.test"},
+            {"http_proxy": "proxy.test", "https_proxy": "proxy.test"},
+            id="http_https_proxy",
+        ),
+    ],
+    indirect=["set_environment_variable"],
+)
+@pytest.mark.usefixtures("set_environment_variable")
+def test_http_proxy(harness: Harness, expected: typing.Dict[str, str]):
```

Comment:
Do we really need harness to unittest this method ? I don't like the habit of having systematically harness to make unitest as harness acts almost like an integration test

Reply:
It indeed doesn't look like harness is needed

Reply:
Removed and using a magic mock instead, thanks!